AlfrescoApp/View Controllers/Sync View Controller/RealmSyncViewController.m

```diff
@@ -0,0 +1,495 @@
+    }
+}
+
+#pragma mark - Private methods
+- (NSString *)listTitle
+{
+    NSString *title = @"";
+    
+    if (self.parentNode)
+    {
+        title = self.parentNode.name;
+    }
+    else
+    {
+        title = NSLocalizedString(@"sync.title", @"Sync Title");
+    }
+    
+    self.emptyMessage = NSLocalizedString(@"sync.empty", @"No Synced Content");
+    return title;
+}
+
+- (void) setupBarButtonItems
+{
+    NSMutableArray *rightBarButtonItems = [NSMutableArray array];
+    
+    // update the UI based on permissions
+    self.switchLayoutBarButtonItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"dots-A"] style:UIBarButtonItemStylePlain target:self action:@selector(showLayoutSwitchPopup:)];
+    
+    [rightBarButtonItems addObject:self.switchLayoutBarButtonItem];
+    
+    [self.navigationItem setRightBarButtonItems:rightBarButtonItems animated:NO];
+}
+
+- (void)showLayoutSwitchPopup:(UIBarButtonItem *)sender
+{
+    [self setupActionsAlertController];
+    self.actionsAlertController.modalPresentationStyle = UIModalPresentationPopover;
+    UIPopoverPresentationController *popPC = [self.actionsAlertController popoverPresentationController];
+    popPC.barButtonItem = self.switchLayoutBarButtonItem;
+    popPC.permittedArrowDirections = UIPopoverArrowDirectionAny;
+    popPC.delegate = self;
+    
+    [self presentViewController:self.actionsAlertController animated:YES completion:nil];
+}
+
+- (void)setupActionsAlertController
+{
+    self.actionsAlertController = [UIAlertController alertControllerWithTitle:nil message:nil preferredStyle:UIAlertControllerStyleActionSheet];
+    
+    NSString *changeLayoutTitle;
+    if(self.style == CollectionViewStyleList)
+    {
+        changeLayoutTitle = NSLocalizedString(@"browser.actioncontroller.grid", @"Grid View");
+    }
+    else
+    {
+        changeLayoutTitle = NSLocalizedString(@"browser.actioncontroller.list", @"List View");
+    }
+    UIAlertAction *changeLayoutAction = [UIAlertAction actionWithTitle:changeLayoutTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
+        if(self.style == CollectionViewStyleList)
+        {
+            [self changeCollectionViewStyle:CollectionViewStyleGrid animated:YES];
+        }
+        else
+        {
+            [self changeCollectionViewStyle:CollectionViewStyleList animated:YES];
+        }
+    }];
+    [self.actionsAlertController addAction:changeLayoutAction];
+    
+    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", @"Cancel") style:UIAlertActionStyleCancel handler:^(UIAlertAction * action) {
+        [self.actionsAlertController dismissViewControllerAnimated:YES completion:nil];
+    }];
+    
+    [self.actionsAlertController addAction:cancelAction];
+}
+
+- (void)showPopoverForFailedSyncNodeAtIndexPath:(NSIndexPath *)indexPath
+{
+//    RealmSyncManager *syncManager = [RealmSyncManager sharedManager];
```

Comment:
This is work in progress - yes, the method is needed and the code will be adapted to the new structure in the next commit.